src/Conjure/Language/Validator.hs

```diff
@@ -0,0 +1,2771 @@
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Conjure.Language.Validator where
```

Comment:
Write a short comment describing what this module does. Which is: mapping from AST to Model, capturing all the errors in the process.